src/backend/joanie/tests/payment/test_backend_base.py

```diff
@@ -447,8 +447,8 @@ def test_payment_backend_base_do_on_payment_failure(self):
         # - Payment has failed gracefully and changed order state to no payment
         self.assertEqual(order.state, enums.ORDER_STATE_NO_PAYMENT)
 
-        # - No email has been sent
-        self.assertEqual(len(mail.outbox), 0)
+        # - An email should be sent mentioning the payment failure
+        self.assertEqual(len(mail.outbox), 1)
```

Comment:
You should something similar to `_check_installment_paid_email_sent` but for a failure to check that an email is sent and the good one is sent.